articles/user-guide/host-programs.md

```diff
@@ -0,0 +1,455 @@
+        }
+    }
+
+    operation MeasureSuperpositionArray(n : Int) : Result[] {
+        mutable resultArray = new Result[n];
+        using (qubits = Qubit[n]) {
+            ApplyToEach( H , qubits);
+            set resultArray = MultiM(qubits);
+            ResetAll(qubits);
+        }
+        return resultArray;
+    }
+}
+```
+
+Select the tab corresponding to your host language of interest.
+
+### [Python](#tab/tabid-python)
+A python host file is constructed as follows:
+1. Import the `qsharp` module, which registers the module loader for Q# interoperability. 
+    This allows Q# namespaces to appear as Python modules, from which we can import Q# operations.
+
+2. Import those Q# operations which we will directly invoke---in this case, `MeasureSuperposition` and `MeasureSuperpositionArray`.
+    ```python
+    import qsharp
+    from NamespaceName import MeasureSuperposition, MeasureSuperpositionArray
+    ```
+    Note that with the `qsharp` module imported, you can also import operations directly from the Q# library namespaces.
+
+3. Among any other Python code, you can now call those operations on specific target machines, and assign their returns to variables (if they return a value) for further use.
+
+#### Specifying target machines
+Calling an operation to be run on a specific target machine is done via different Python methods on the imported operation.
+For example, `.simulate(<args>)`, uses the `QuantumSimulator` to run the operation, whereas `.estimate_resources(<args>)` does so on the `ResourcesEstimator`.
+
+#### Arguments
+Arguments for the Q# callable should be provided in the form of a keyword argument, where the keyword is the argument name in the callable definition.
+That is, `MeasureSuperpositionArray.simulate(n=4)` is valid, whereas `MeasureSuperpositionArray.simulate(4)` would throw an error.
+
+Therefore, the Python host program 
+
+```python
+import qsharp
+from NamespaceName import MeasureSuperposition, MeasureSuperpositionArray
+
+single_qubit_result = MeasureSuperposition.simulate()
+single_qubit_resources = MeasureSuperposition.estimate_resources()
+
+multi_qubit_result = MeasureSuperpositionArray.simulate(n=4)
+multi_qubit_resources = MeasureSuperpositionArray.estimate_resources(n=4)
+
+print('Single qubit:')
+print(single_qubit_result)
+print(single_qubit_resources)
+
+print('\nMultiple qubits:')
+print(multi_qubit_result)
+print(multi_qubit_resources)
+```
+
+results in an output like the following:
+
+```python
+Single qubit:
+1
+{'CNOT': 0, 'QubitClifford': 1, 'R': 0, 'Measure': 1, 'T': 0, 'Depth': 0, 'Width': 1, 'BorrowedWidth': 0}
+
+Multiple qubits:
+[1, 1, 0, 0]
+{'CNOT': 0, 'QubitClifford': 4, 'R': 0, 'Measure': 8, 'T': 0, 'Depth': 0, 'Width': 4, 'BorrowedWidth': 0}
+```
+
+### [C#](#tab/tabid-csharp)
+
+A C# host program has multiple components, and works very closely with some components of the QDK, such as the simulators, which are also built on the .NET language.
```

Comment:
Strictly speaking, there's no such thing as "the .NET language," as .NET is intended to be a multi-language platform (indeed, .NET is also known as the CLR, or _common language runtime_).